### PR TITLE
[docs] [ISSUE 11625] Update the intro of Messaging doc

### LINKED ISSUE #11625: [doc] Message redelivery and acknowledgement
**Is your feature request related to a problem? Please describe.**
In the current [messaging docs](https://pulsar.apache.org/docs/en/concepts-messaging/), there is no content related to message redelivery and acknowledgement.

**Describe the solution you'd like**
Add clear and concise description of message redelivery and acknowledgement.

### PR DESCRIPTION
**Pull request by sijia-w**
Fixes #11625

### CODE REVIEW

**Comment by sijia-w** [comment]
@Huanli-Meng @Jennifer88huang Please help review the changes. If more information should be provided, please leave a comment. Thank you.

File: site2/docs/concepts-messaging.md [hunk 1]
```diff
@@ -180,18 +182,18 @@ If you want to acknowledge messages negatively, you can use the following API.
...
 
-> **Note**
-> If batching is enabled, other messages and the negatively acknowledged messages in the same batch are redelivered to the consumer.
+> **Note**  
+> If batching is enabled, all messages in one batch are redelivered to the consumer.
 
 ### Acknowledgement timeout
 
-If a message is not consumed successfully, and you want to trigger the broker to redeliver the message automatically, you can adopt the unacknowledged message automatic re-delivery mechanism. Client tracks the unacknowledged messages within the entire `acktimeout` time range, and sends a `redeliver unacknowledged messages` request to the broker automatically when the acknowledgement timeout is specified.
+If a message is not consumed successfully, and you want the broker to redeliver this message automatically, then you can adopt the unacknowledged message automatic redelivery mechanism. With this redelivery approach enabled, the client tracks the unacknowledged messages within the entire `acktimeout` time range, and sends a `redeliver unacknowledged messages` request to the broker automatically when the acknowledgement timeout is specified.
```
Comment (Huanli-Meng) [inline site2/docs/concepts-messaging.md:190]:
```suggestion
If a message is not consumed successfully, and you want the broker to redeliver this message automatically, then you can enable automatic redelivery mechanism for  unacknowledged messages. With automatic redelivery enabled, the client tracks the unacknowledged messages within the entire `acktimeout` time range, and sends a `redeliver unacknowledged messages` request to the broker automatically when the acknowledgement timeout is specified.
```

**Review by Huanli-Meng — APPROVED** [review]
Overall LGTM. only leave 2 minor comments.

**Comment by sijia-w** [comment]
@Huanli-Meng Thanks for your comments. Updated. PTAL.

**Pull request merged**